crates/uv-python/src/sysconfig/cursor.rs

```diff
@@ -0,0 +1,148 @@
+#![allow(dead_code)]
+
+use std::str::Chars;
+
+pub(super) const EOF_CHAR: char = '\0';
+
+/// A cursor represents a pointer in the source code.
+///
+/// Based on [`rustc`'s `Cursor`](https://github.com/rust-lang/rust/blob/d1b7355d3d7b4ead564dbecb1d240fcc74fff21b/compiler/rustc_lexer/src/cursor.rs)
```

Comment:
This looks a lot like https://github.com/typst/unscanny

Reply:
That looks interesting!